src/mscorlib/shared/System/Span.NonGeneric.cs

```diff
@@ -259,13 +259,7 @@ internal static unsafe void CopyTo<T>(ref T destination, ref T source, int eleme
             nuint byteCount = (nuint)elementsCount * (nuint)Unsafe.SizeOf<T>();
```

Comment:
Are the special cases for single element and `Unsafe.AreSame` still worth it? I think we may be a good idea to move the AreSame calls into Memmove, not have the special case for single element; and maybe get rid of the `CopyTo<T>` helper by manually inlining it into the few callsites where it is used.